operator/pkg/ingress/annotations/annotations.go

```diff
@@ -185,3 +187,53 @@ func GetAnnotationTLSPassthroughEnabled(ingress *networkingv1.Ingress) bool {
 
 	return boolVal
 }
+
+// GetAnnotationEnforceHTTPSEnabled retrieves the EnforceHTTPS annotation's value.
+// This uses a string rather than a bool because the empty string means "unset".
+// In this case this matters because if the value is unset, it can be overridden
+// by the global config option `--enforce-ingress-https`.
+//
+// If the annotation is set, will override the global config option in all cases.
+//
+// Note that `enabled`, `disabled` and `true` or `false` style values (as understood by
+// strconv.ParseBool() ) will work. The annotation being present but set to any
+// other value will result in returning the empty string (as in, the same as if
+// unset).
+//
+// If the annotation is unset, this returns "" (the empty string).
+//
+// The only valid values are:
+// - "true" - the annotation is present and set to a truthy value
+// - "false" - the annovation is present and set to a false value
+// - "" - the annotatation is not present
+func GetAnnotationForceHTTPSEnabled(ingress *networkingv1.Ingress) *bool {
+	returnBool := new(bool)
+
+	val, exists := annotation.Get(ingress, ForceHTTPSAnnotation)
+	if !exists {
+		return nil
+	}
+
+	if val == enabled {
+		*returnBool = true
+		return returnBool
```

Comment:
you can use `model.AddressOf(true)` - this way we can get rid of the variable `returnBool` -> https://github.com/cilium/cilium/blob/main/operator/pkg/model/helpers.go#L89